File: Store.Web/src/app/models/Cheese.decorator.ts
import { IPizza } from "./IPizza";
import { PizzaDecorator } from "./Pizza.decorator";

export class CheeseDecorator extends PizzaDecorator {
    private cheeseImageUrl: string;

    constructor(override inputPizza: IPizza) {
        super(inputPizza);
        this.cheeseImageUrl = 'assets/Images/cheese.png'; // URL for the cheese image
    }

    override GetIngredients(): string {
        return super.GetIngredients() + ' cheese';   
    }

    override GetCost(): number {
        return super.GetCost() + 5;
    }
    
    override GetImageUrl(): string {
        return super.GetImageUrl() + ',' + this.cheeseImageUrl; // Concatenate image URLs
    }
}

File: StoreApi/Decorators/CheeseDecorator.cs
using StoreApi.Models;

namespace StoreApi.Decorators
{
    public class CheeseDecorator : PizzaDecorator
    {
        public CheeseDecorator(IPizza pizza) : base(pizza) { }

        public  string GetDescription() => _pizza.GetDescription() + ", Cheese";
        public override double GetCost() => _pizza.GetCost() + 1.25;
    }

}
